### PR TITLE
Fix typeflag reads in tgz.go

### LINKED ISSUE #889: Installing helm chart from custom Helm repository retrieves 0 files
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

With a custom Helm chart repository `myrepo` containing a chart `foo`, `ks pkg install myrepo/foo` retrieves 0 files. This breaks my application as the app.yaml file is modified to include the installed chart as a library, but nothing has been downloaded.

**What you expected to happen**:

My custom chart gets properly installed in my ksonnet application.

**How to reproduce it (as minimally and precisely as possible)**:

1) Custom Helm Chart repository setup in S3 and reachable from the machine.
2) `ks repository add myrepo <repo URL>`
3) `ks pkg install myrepo/foo`

**Anything else we need to know?**:

**Environment**:
- ksonnet version (use `ks version`):
```
ksonnet version: dev-
jsonnet version: v0.11.2
client-go version:
```
(built from source from the v0.13.1 branch)
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"", BuildDate:"2018-10-29T07:50:18Z", GoVersion:"go1.11.1", Compiler:"gc", Platform:"freebsd/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.3-eks", GitCommit:"58c199a59046dbf0a13a387d3491a39213be53df", GitTreeState:"clean", BuildDate:"2018-09-21T21:00:04Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
Closes #889 

Signed-off-by: GuessWhoSamFoo <sfoohei@gmail.com>

### CODE REVIEW

**Comment by sh0rez** [comment]
Can this please be merged? Without this fix one of ksonnet's most important features is only available on osx ..

**Comment by kvaps** [comment]
shipit

File: pkg/util/archive/tgz.go [hunk 1]
```diff
@@ -62,6 +65,8 @@ func (t *Tgz) Unarchive(r io.Reader, handler FileHandler) error {
 			if err = handler(tf); err != nil {
 				return err
 			}
+		default:
+			return fmt.Errorf("Unknown header typeflag:  %v", header.Typeflag)
```
Comment (shomron) [inline pkg/util/archive/tgz.go]: There's a whole bunch of other valid (but skippable) typeflags - I don't think we need to error out if encountering them. Perhaps remove the `default` and just handle the types we care about?

Comment (GuessWhoSamFoo) [reply pkg/util/archive/tgz.go]: Removed the case for TypeDir as well since that was explicitly skipped previously and rebased :+1:

**Pull request merged**